plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1560,7 +1569,8 @@ bool producer_plugin_impl::remove_expired_trxs( const fc::time_point& deadline )
                if( pbm == pending_block_mode::producing ) {
                   fc_dlog( _trx_trace_log,
                            "[TRX_TRACE] Block ${block_num} for producer ${prod} is EXPIRING PERSISTED tx: ${txid}",
-                           ("block_num", chain.head_block_num() + 1)("prod", chain.pending_block_producer())("txid", txid));
+                           ("block_num", chain.head_block_num() + 1)("txid", txid)
+                           ("prod", chain.is_building_block() ? chain.pending_block_producer() : name()) );
```

Comment:
Is there a reason you don't use the new method here as well?

Reply:
impl this not available and I didn't want to have to calculate it and pass it in when most of the time logging is off and it is not needed.